packages/mobile/src/components/CurrencyDisplay.tsx

```diff
@@ -180,7 +181,9 @@ export default function CurrencyDisplay({
   const sign = value?.isNegative() ? '-' : showExplicitPositiveSign ? '+' : ''
   const formatAmount = getFormatFunction(formatType)
   const formattedValue =
-    value && displayCurrency ? formatAmount(value.absoluteValue(), displayCurrency) : '-'
+    value && displayCurrency && value.lt(WALLET_BALANCE_UPPER_BOUND)
+      ? formatAmount(value.absoluteValue(), displayCurrency)
+      : '-'
```

Comment:
I would rather not have this check here.
I think this is too defensive and could be surprising in the future, given that incorrect high values are handled in the tokens saga (where they can happen).